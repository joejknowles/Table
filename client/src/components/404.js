import React from 'react';
import res from '../resources/pages/404';

export default () => (
  <div className="not-found-message">
    { res.message }
  </div>
);
