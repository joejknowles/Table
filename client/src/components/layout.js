import React from 'react';

import Notifications from './notifications';

export default ({ children }) => (
  <div className="App">
    { children }
    <Notifications />
  </div>
);
