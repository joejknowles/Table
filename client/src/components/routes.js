import React from 'react';
import { Router, Route, browserHistory } from 'react-router';
import StartScreen from './startScreen';
import Waiting from './waiting';
import Play from './play';
import Table from './table';
import NotFound from './404';
import paths from '../shared/paths';

export default () => (
  <Router history={ browserHistory }>
    <Route path={ paths.startScreen } component={ StartScreen } />
    <Route path={ paths.play + '(/:gameCode)' } component={ Play } />
    <Route path={ paths.table } component={ Table } />
    <Route path={ paths.waiting } component={ Waiting } />
    <Route path='/*' component={ NotFound } />
  </Router>
);
