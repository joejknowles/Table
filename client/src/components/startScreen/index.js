import React from 'react';

import Layout from '../layout'
import JoinGame from './joinGame';
import NewGame from './newGame';

export default () => (
  <Layout>
    <JoinGame />
    <hr />
    <NewGame />
  </Layout>
);
