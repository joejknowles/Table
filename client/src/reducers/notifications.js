export const notification = (state = '', action) => {
  switch (action.type) {
    case 'ADD_NOTIFICATION':
      return action.message;
    case 'REMOVE_NOTIFICATION':
      return '';
    default:
      return state;
  }
};
