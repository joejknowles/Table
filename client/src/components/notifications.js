import React from 'react';
import { connect } from 'react-redux';
import { notificationSelector } from '../reducers';

export const Notifications = ({ notification }) => (
  notification ? <p>{ notification }</p> : null
);

const mapStateToProps = (state) => ({
  notification: notificationSelector(state)
});

export default connect(mapStateToProps)(Notifications);
