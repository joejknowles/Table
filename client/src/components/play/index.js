import React from 'react';
import { connect } from 'react-redux';

import Hand from './hand';
import { handSelector } from '../../reducers';
import '../../styles/Play.css';
import res from '../../resources/pages/play';

export const Play = ({ hand }) => (
  <div className="App Play">
    <div className="deck">
      { hand > 0 ?
          <Hand /> :
          <p className='no-cards-message'>
            { res.noCardsMessage }
          </p>
      }
    </div>
  </div>
);


export default connect(
  (state) => ({ hand: handSelector(state) })
)(Play);
