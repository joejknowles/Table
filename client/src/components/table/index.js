import React from 'react';
import { connect } from 'react-redux';

import Layout from '../layout';
import { tablePileTopCardSelector } from '../../reducers';

export const Table = ({ topCard }) => (
  <Layout>
    <div className="pile">
      { topCard ?
        <div >
          <img className="card" src={ topCard.img } />
        </div> : <div className="noCards" /> }
    </div>
  </Layout>
);

const mapStateToProps = (state) => ({
  topCard: tablePileTopCardSelector(state)
});

export default connect(mapStateToProps)(Table);
