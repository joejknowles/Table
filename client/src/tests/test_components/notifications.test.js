import { Notifications } from '../../components/notifications';
import React from 'react';
import ReactDOM from 'react-dom';

import ReactTestUtils from 'react-addons-test-utils';
const shallowRenderer = ReactTestUtils.createRenderer();

it('component renders as before with notifications', () => {
  const component = shallowRenderer.render(
    <Notifications
      notification='test notification'
    />
  );
  expect(component).toMatchSnapshot();
});

it('component renders as before without notifications', () => {
  const component = shallowRenderer.render(
    <Notifications
      notification=''
    />
  );
  expect(component).toMatchSnapshot();
});
