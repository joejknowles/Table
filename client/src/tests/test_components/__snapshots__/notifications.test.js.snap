exports[`test component renders as before with notifications 1`] = `
<p>
  test notification
</p>
`;

exports[`test component renders as before without notifications 1`] = `null`;
